sdk/src/message.rs

```diff
@@ -57,11 +57,23 @@ impl InstructionKeys {
     }
 }
 
+/// Return the pubkey of the first writable signer in the given set of instructions.
+pub fn get_first_writable_signer(instructions: &[Instruction]) -> Option<&Pubkey> {
+    for instruction in instructions {
```

Comment:
It may not match the payer logic in load_accounts that determines the payer. I'm not sure if that's a problem though.

Reply:
Nevermind, I see it's just used for the construction so it should be fine.